### ISSUE TITLE
[graphShapeAnalyzer.cpp::processCheck::587] Error Code 4: Internal Error (Concat_151: dimensions not compatible for concatenation)

### ISSUE THREAD

**Issue by StevenChen16** [NVIDIA/TensorRT#2731]
## Description

When I use trtexec to turn .onnx file into .engine file ,there's something went wrong.
It told me that:
"""
[03/04/2023-20:16:04] [W] [TRT] onnx2trt_utils.cpp:369: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[03/04/2023-20:16:05] [I] Finish parsing network model
[03/04/2023-20:16:05] [W] Dynamic dimensions required for input: images, but no shapes were provided. Automatically overriding shape to: 1x3x1x1
[03/04/2023-20:16:05] [E] Error[4]: [graphShapeAnalyzer.cpp::nvinfer1::builder::`anonymous-namespace'::ShapeNodeRemover::processCheck::587] Error Code 4: Internal Error (Concat_151: dimensions not compatible for concatenation)
[03/04/2023-20:16:05] [E] Error[2]: [builder.cpp::nvinfer1::builder::Builder::buildSerializedNetwork::636] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
[03/04/2023-20:16:05] [E] Engine could not be created from network
[03/04/2023-20:16:05] [E] Building engine failed
[03/04/2023-20:16:05] [E] Failed to create engine from model or file.
[03/04/2023-20:16:05] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8403] # D:\ProgramData\TensorRT-8.4.3.1\bin\trtexec.exe --onnx=best.onnx --saveEngine=best.engine --tacticSources=-cublasLt,+cublas
"""

## Environment

**TensorRT Version**: **8.4.3.1**
**NVIDIA GPU**: **MX350**
**NVIDIA Driver Version**: **527.56**
**CUDA Version**: **11.6**
**CUDNN Version**: **8.7.0**
**Operating System**: Windows 11
**Python Version (if applicable)**:3.10.8 
**Tensorflow Version (if applicable)**: 2.10.1
**PyTorch Version (if applicable)**: 1.12.0+cu11

**Comment by zerollzeng** [comment]
Looks like something wrong with your onnx model or an internal bug for TRT, can you try to run it with onnxruntime first? if it works with onnxruntime then it's a bug in TRT, then can you provide the onnx to us? Thanks!

**Comment by hong-kailin** [comment]
Hello, I have the same problem, and I have no problem using onnxruntime for reasoning, but using tensorRT will report an error.

`[03/08/2023-11:03:25] [E] Error[4]: [shapeCompiler.cpp::nvinfer1::builder::DynamicSlotBuilder::evaluateShapeChecks::1180] Error Code 4: Internal Error (kOPT values for profile 0 violate shape constraints: Concat_1443: dimensions not compatible for concatenation Condition '==' violated: 50 != 25.)
[03/08/2023-11:03:25] [E] Error[2]: [builder.cpp::nvinfer1::builder::Builder::buildSerializedNetwork::751] Error Code 2: Internal Error (Assertion engine != nullptr failed. )`

**Comment by hong-kailin** [comment]
![image](https://user-images.githubusercontent.com/40348459/223610856-a5609d32-7fcb-4e01-a6d5-dfc3a516458e.png)

**Comment by zerollzeng** [comment]
Looks like a bug in TRT, can you provide the onnx model to us? or a sub onnx graph that can reproduce this issue. Thanks！

**Comment by StevenChen16** [comment]
I've run it on onnxruntime successfully, but i can't run it on TRT. I've also tried to turn the yolov5s.onnx into yolov5s.engine  successfully,but I can't use TRT to deal with the yolov8m.onnx.It might be a bug in TRT

**Comment by zerollzeng** [comment]
Can you share the onnx model here? Thanks！We can check it internally to see what happens.

**Comment by hong-kailin** [comment]
Hello, I made another attempt two days ago: remove some operators, and I want to see which operator is out. But after I removed some operators, I found another problem. I guess this problem may be the cause of the above problem. But the error of this problem reminds me that I haven't found the reason.

**Comment by hong-kailin** [comment]
Hello, I made another attempt two days ago: remove some operators, and I want to see which operator is out. But after I removed some operators, I found another problem. I guess this problem may be the cause of the above problem. But the error of this problem reminds me that I haven't found the reason.

【Error message】：
```
[03/13/2023-11:25:12] [E] Error[2]: [runner.cpp::nvinfer1::rt::MyelinRunnerBase::retrieveOutputTensorResult::608] Error Code 2: Internal Error (Assertion actualSize <= size failed. )
[03/13/2023-11:25:12] [E] Error[2]: [builder.cpp::nvinfer1::builder::Builder::buildSerializedNetwork::751] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
```

【TRT version】：8.5.1.7

**Comment by hong-kailin** [comment]
Hello, I made another attempt: remove some operators, and I want to see which operator caused the error. But after I removed some operators, I found another problem. I guess this problem may be the cause of the above problem. But I haven't found the reason.

【Error message】:
```
trtexec.exe --onnx=end2end.sim.onnx --saveEngine=output.engine

[03/13/2023-11:25:12] [E] Error[2]: [runner.cpp::nvinfer1::rt::MyelinRunnerBase::retrieveOutputTensorResult::608] Error Code 2: Internal Error (Assertion actualSize <= size failed. )
[03/13/2023-11:25:12] [E] Error[2]: [builder.cpp::nvinfer1::builder::Builder::buildSerializedNetwork::751] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
```
【TRT version】:8.5.1.7
【model】: https://drive.google.com/file/d/1kR5OwewHr1ZYPfNCbfmIy8aQ1gLNtmNn/view?usp=share_link

**Comment by zerollzeng** [comment]
I've request access.

**Comment by hong-kailin** [comment]
Sorry, it can be downloaded now.

**Comment by hong-kailin** [comment]
> I've requested access.
Sorry, it can be downloaded now.

**Comment by hong-kailin** [comment]
Sorry, it can be downloaded now.

**Comment by zerollzeng** [comment]
Did a quick test, this should be fixed at 8.6 GA
```
[03/16/2023-07:18:58] [I] === Performance summary ===
[03/16/2023-07:18:58] [I] Throughput: 82.9246 qps
[03/16/2023-07:18:58] [I] Latency: min = 11.9569 ms, max = 12.2188 ms, mean = 12.0319 ms, median = 12.0171 ms, percentile(90%) = 12.119 ms, percentile(95%) = 12.1519 ms, percentile(99%) = 12.1818 ms
[03/16/2023-07:18:58] [I] Enqueue Time: min = 11.9467 ms, max = 12.1924 ms, mean = 12.0189 ms, median = 12.0049 ms, percentile(90%) = 12.1017 ms, percentile(95%) = 12.1342 ms, percentile(99%) = 12.1624 ms
[03/16/2023-07:18:58] [I] H2D Latency: min = 0.258789 ms, max = 0.277222 ms, mean = 0.260515 ms, median = 0.259766 ms, percentile(90%) = 0.262695 ms, percentile(95%) = 0.265625 ms, percentile(99%) = 0.270386 ms
[03/16/2023-07:18:58] [I] GPU Compute Time: min = 11.6917 ms, max = 11.9402 ms, mean = 11.7637 ms, median = 11.7495 ms, percentile(90%) = 11.848 ms, percentile(95%) = 11.8816 ms, percentile(99%) = 11.9108 ms
[03/16/2023-07:18:58] [I] D2H Latency: min = 0.00558472 ms, max = 0.0291748 ms, mean = 0.00769299 ms, median = 0.00683594 ms, percentile(90%) = 0.0100098 ms, percentile(95%) = 0.013916 ms, percentile(99%) = 0.0187988 ms
[03/16/2023-07:18:58] [I] Total Host Walltime: 3.01479 s
[03/16/2023-07:18:58] [I] Total GPU Compute Time: 2.94091 s
[03/16/2023-07:18:58] [W] * Throughput may be bound by Enqueue Time rather than GPU Compute and the GPU may be under-utilized.
[03/16/2023-07:18:58] [W]   If not already in use, --useCudaGraph (utilize CUDA graphs where possible) may increase the throughput.
[03/16/2023-07:18:58] [I] Explanations of the performance metrics are printed in the verbose logs.
[03/16/2023-07:18:58] [V]
[03/16/2023-07:18:58] [V] === Explanations of the performance metrics ===
[03/16/2023-07:18:58] [V] Total Host Walltime: the host walltime from when the first query (after warmups) is enqueued to when the last query is completed.
[03/16/2023-07:18:58] [V] GPU Compute Time: the GPU latency to execute the kernels for a query.
[03/16/2023-07:18:58] [V] Total GPU Compute Time: the summation of the GPU Compute Time of all the queries. If this is significantly shorter than Total Host Walltime, the GPU may be under-utilized because of host-side overheads or data transfers.
[03/16/2023-07:18:58] [V] Throughput: the observed throughput computed by dividing the number of queries by the Total Host Walltime. If this is significantly lower than the reciprocal of GPU Compute Time, the GPU may be under-utilized because of host-side overheads or data transfers.
[03/16/2023-07:18:58] [V] Enqueue Time: the host latency to enqueue a query. If this is longer than GPU Compute Time, the GPU may be under-utilized.
[03/16/2023-07:18:58] [V] H2D Latency: the latency for host-to-device data transfers for input tensors of a single query.
[03/16/2023-07:18:58] [V] D2H Latency: the latency for device-to-host data transfers for output tensors of a single query.
[03/16/2023-07:18:58] [V] Latency: the summation of H2D Latency, GPU Compute Time, and D2H Latency. This is the latency to infer a single query.
[03/16/2023-07:18:58] [I]
&&&& PASSED TensorRT.trtexec [TensorRT v8600] # trtexec --onnx=end2end.sim.onnx --verbose
```

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thank you!
